datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -20,35 +24,45 @@
 
     private final BookModelAssembler assembler;
 
+
+
     public BookController(BookRepository repository, BookModelAssembler assembler) {
         this.repository = repository;
         this.assembler = assembler;
     }
 
-    @PostMapping("/books")
-    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book){
+    @PostMapping("/bookstores/{store_id}/books")
+    protected ResponseEntity<EntityModel<Book>> newBook(@RequestBody Book book, @PathVariable Long store_id){
+        book.setStore_id(store_id);
         EntityModel<Book> entityModel = assembler.toModel(repository.save(book));
+
         return ResponseEntity
                 .created(entityModel.getRequiredLink(IanaLinkRelations.SELF).toUri())
                 .body(entityModel);
     }
     
-    @GetMapping("/books/{id}")
-    protected EntityModel<Book> one(@PathVariable Long id) {
+    @GetMapping("/bookstores/{store_id}/books/{id}")
+    protected EntityModel<Book> one(@PathVariable Long id, @PathVariable Long store_id) {
         Book book = repository.findById(id).orElseThrow(() -> new BookNotFoundException(id));
+
         return assembler.toModel(book);
+
     }
 
-    @GetMapping("/books")
-    protected CollectionModel<EntityModel<Book>> all(){
-        List<EntityModel<Book>> books = repository.findAll().stream()
+    @GetMapping("/bookstores/{store_id}/books")
+    protected CollectionModel<EntityModel<Book>> all(@PathVariable long store_id){
+        List<Book> books = repository.findAll();
+        books.removeIf(book -> book.getStore_id() == null || book.getStore_id() != store_id);
+
+        List<EntityModel<Book>> booksAll = books
+               .stream()
                 .map(assembler::toModel)
-                .collect(Collectors.toList());
-        return CollectionModel.of(books, linkTo(methodOn(BookController.class).all()).withSelfRel());
+               .collect(Collectors.toList());
+        return CollectionModel.of(booksAll, linkTo(methodOn(BookController.class).all(store_id)).withSelfRel());
     }
 
-    @PutMapping("/books/{id}")
-    protected Book updateBook(@RequestBody Book newBook, @PathVariable Long id) {
+    @PutMapping("/bookstores/{store_id}/books/{id}")
+    protected Book updateBook(@RequestBody Book newBook, @PathVariable Long id, @PathVariable Long store_id) {
```

Comment:
Same question here - what if the store doesnt exist or if it doesnt match the book?